### PR TITLE
Add support for fp16 in convnet_benchmarks.py

### PR DESCRIPTION
**Pull request by sh1r0**

### CODE REVIEW

File: caffe2/python/convnet_benchmarks.py [hunk 1]
```diff
@@ -80,25 +82,32 @@ def MLP(order, cudnn_ws):
                 dim_out=d,
                 weight_init=('XavierFill', {}),
                 bias_init=('XavierFill', {}),
+                WeightInitializer=initializer,
```
Comment (harouwu) [inline caffe2/python/convnet_benchmarks.py]: Nice use of Initializer. But I think you can use `arg_scope` in `brew` module to make the whole code cleaner. PR #1093 is a good example of it.

File: caffe2/python/convnet_benchmarks.py [hunk 2]
```diff
@@ -80,25 +82,32 @@ def MLP(order, cudnn_ws):
...
         "last",
         dim_in=d,
         dim_out=1000,
         weight_init=('XavierFill', {}),
         bias_init=('XavierFill', {}),
+        WeightInitializer=initializer,
+        BiasInitializer=initializer
     )
-    xent = model.net.LabelCrossEntropy(["last", "label"], "xent")
+    if use_fp16:
```
Comment (harouwu) [inline caffe2/python/convnet_benchmarks.py]: Do you think it is possible to refactor all of these Loss part so that our model building function can be dtype unaware?

**Review by harouwu — CHANGES_REQUESTED** [review]
Nice code! fp16 is super useful as an example of Caffe2. one major change request is to use`brew.arg_scope` to make code cleaner. Please check out caffe2.ai tutorial section: brewing your model. And PR #1093 is also a good example. 👍

**Comment by sh1r0** [comment]
@harouwu I've updated the commit for `arg_scope`. For the 2nd comment, I think it would be nice if `LabelCrossEntropy`, `Softmax`, etc. operations could be dtype aware (probably implicit dtype conversion).

**Comment by orionr** [comment]
Thank you for your contribution! PyTorch and Caffe2 are now officially merged with the PyTorch 1.0 preview release. Please submit on https://github.com/pytorch/pytorch if this is still an issue over there.
